mmv1/products/billingbudget/api.yaml

```diff
@@ -29,11 +29,9 @@ objects:
   - !ruby/object:Api::Resource
     name: Budget
     base_url: billingAccounts/{{billing_account}}/budgets
-    self_link: '{{name}}'
+    self_link: 'billingAccounts/{{billing_account}}/budgets/{{name}}'
```

Comment:
I don't think this will be backwards compatible.

If a user has `name` in state as the long form name as it would be stored today, and they update to this version of the provider they won't be able to fetch the resource because the URL they call will be `'billingAccounts/{{billing_account}}/budgets/{{long_name}}'`

We may need a state migrate function similar to: https://github.com/GoogleCloudPlatform/magic-modules/pull/4669

Reply:
Thanks! I wasn't sure how to address that. I've added a state migration function in ef2f5c9